### ISSUE TITLE
AnimatedSvgView not display center on the screen

### ISSUE THREAD

**Issue by Vishal12796** [jaredrummler/AnimatedSvgView#6]
AnimatedSvgView not display center on the screen

``` xml
<FrameLayout    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="fill_parent"
    android:id="@+id/rl"
    android:background="@mipmap/back2" >

    <RelativeLayout
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_gravity="center"
        android:layout_centerVertical="true"
        android:layout_centerHorizontal="true" >

        <com.jrummyapps.android.widget.AnimatedSvgView
            android:id="@+id/animated_svg_view"
            android:layout_width="100dp"
            android:layout_height="100dp"
            android:layout_gravity="center"
            app:paddingEnd="@dimen/activity_horizontal_margin"
            app:animatedSvgFillColors="@array/os_colors"
            app:animatedSvgGlyphStrings="@array/os_strings"
            app:animatedSvgImageSizeX="400"
            app:animatedSvgImageSizeY="400"
            app:animatedSvgFillStart="1200"
            app:animatedSvgFillTime="1000"
            app:animatedSvgTraceTime="2000"
            app:animatedSvgTraceTimePerGlyph="1000" />

    </RelativeLayout>

</FrameLayout>
```

**Comment by jaredrummler** [comment]
Will you please add the arrays?

**Comment by Vishal12796** [comment]
``` xml
<string-array name="os_strings">

        <item>M68,200.7c-9.9,0 -18.8,-2.1 -26.7,-6.1c-8,-4 -14.6,-9.7 -20.2,-16.9c-5.6,-7.3 -9.8,-15.9 -12.9,-26c-2.9,-10 -4.5,-21.1 -4.5,-33.1c0,-6.3 0.4,-12.9 1.4,-19.5C6,92.4 7.4,85.6 9.4,79c1.9,-6.7 4.3,-13.2 7.1,-19.5c2.8,-6.4 6.1,-12.4 9.8,-18.1c3.7,-5.7 8,-10.9 12.6,-15.7c4.7,-4.8 9.7,-9 15.4,-12.4c5.6,-3.5 11.7,-6.2 18.1,-8.2S85.8,2 93.2,2c9.9,0 18.9,1.7 26.7,5.3s14.5,8.6 19.9,15.1c5.3,6.4 9.5,14.1 12.4,23c2.8,9 4.4,18.8 4.4,29.5c0,8.8 -0.9,18.1 -2.6,27.8c-1.7,9.6 -4.4,19.1 -7.8,28.5c-3.4,9.3 -7.6,18.2 -12.7,26.6c-5,8.4 -10.8,15.8 -17.4,22.2c-5.2,5.1 -11,9.4 -17.3,12.8c-1.5,0.9 -3.1,1.6 -4.6,2.3C86.1,198.9 77.4,200.7 68,200.7zM70.4,176.5c5.3,0 10.3,-1 14.8,-3.1c4.6,-2.1 8.7,-4.9 12.6,-8.4c3.8,-3.6 7.3,-7.8 10.5,-12.6c3.2,-4.8 6,-9.9 8.4,-15.5c2.4,-5.5 4.6,-11.2 6.4,-17.1s3.4,-11.8 4.5,-17.5c1.2,-5.8 2.1,-11.4 2.6,-16.7c0.5,-5.3 0.9,-10.2 0.9,-14.5c0,-6.3 -0.7,-12.3 -2.1,-17.9s-3.5,-10.5 -6.3,-14.6s-6.3,-7.4 -10.6,-9.8c-4.3,-2.4 -9.3,-3.6 -15,-3.6c-6.1,0 -12,1.5 -17.7,4.5c-5.7,2.9 -11,6.9 -16.2,11.9c-5,5 -9.6,10.8 -13.9,17.4c-4.1,6.6 -7.8,13.5 -10.7,20.9c-2.9,7.3 -5.3,14.7 -7,22.3c-1.6,7.5 -2.5,14.7 -2.5,21.7c0,7.5 1,14.5 2.8,21c1.9,6.4 4.6,12 8.2,16.8c3.5,4.7 7.9,8.4 12.9,11.1C58.2,175.2 63.9,176.5 70.4,176.5z</item>

        <item>M144.1,205c-1,-3.5 -2.5,-6.3 -4.6,-8.6c-2.1,-2.3 -4.5,-4.3 -7.3,-5.8c-2.2,-1.2 -3.6,-1.7 -9.2,-3.7c-8.8,-3.1 -9.8,-3.4 -11.2,-2.8c-0.2,0.1 -2.1,1.4 -5.7,4.1c-2,1.5 -2.6,2 -3.8,2.6c-2.4,1.3 -3.4,1.4 -3.4,1.9c0,1 4.4,1.4 12.2,3.2c2.7,0.7 5.6,1.3 8.8,2.6c2.6,1.1 4.9,2.4 6.9,3.9c2,1.5 3.2,3.6 3.8,6.1c0.8,2.7 0.7,5.2 -0.3,7.8c-1,2.5 -2.4,4.8 -4.4,6.9c-2,2.1 -4.5,3.9 -7.3,5.5c-2.8,1.5 -5.8,2.7 -8.8,3.6c-2.6,0.8 -5.1,1.1 -7.8,1.2c-2.5,0.1 -4.9,-0.2 -7.1,-0.8c-2.2,-0.7 -4,-1.7 -5.7,-3.2c-1.6,-1.4 -2.7,-3.4 -3.4,-5.9c-0.3,-1.1 -0.5,-2.2 -0.7,-3.2c-0.1,-1 -0.2,-1.9 -0.2,-2.7c-0.1,-0.9 -0.1,-1.6 -0.1,-2.3s-0.1,-1.3 -0.3,-2c-0.3,-1.3 -1,-2.2 -2,-2.5c-0.9,-0.4 -2,-0.4 -3.1,-0.1c-1.1,0.3 -2.2,0.9 -3.3,1.9c-1.1,0.9 -2.1,2.1 -2.7,3.5c-0.8,1.4 -1.3,3.1 -1.5,5c-0.2,1.9 -0.1,3.9 0.5,6.2c1.2,4.4 3.2,7.9 6,10.6c2.8,2.7 6.1,4.7 9.9,5.9c3.8,1.3 7.9,1.9 12.3,1.7c4.4,-0.1 8.8,-0.7 13.2,-1.9c3.1,-0.9 6.1,-2 9.1,-3.4s5.7,-2.9 8.2,-4.8c2.5,-1.9 4.7,-3.8 6.7,-6.1c2,-2.3 3.5,-4.7 4.7,-7.3c1.2,-2.6 2,-5.3 2.2,-8.3C145.2,211.1 145,208.1 144.1,205z</item>

        <item>M133.5,156.3c-0.9,-3.4 -2.6,-6.2 -5,-8.5c-2.4,-2.3 -5.2,-4 -8.5,-5.3c-3.3,-1.2 -6.8,-2 -10.6,-2.1c-3.8,-0.1 -7.4,0.3 -11.1,1.3c-5.7,1.5 -10.5,3.6 -14.4,6.2c-3.9,2.6 -6.9,5.3 -9.2,8.4c-2.2,3.1 -3.6,6.2 -4.3,9.4c-0.7,3.3 -0.5,6.3 0.2,9.3c0.1,0.4 0.2,0.9 0.4,1.2c0,0.1 0.1,0.2 0.1,0.3c0.3,0 0.5,-0.1 0.9,-0.1c0.2,0 1.3,-0.1 2.7,-0.3c1.4,-0.2 3.2,-0.4 4.6,-0.7c1.3,-0.2 3.8,-1 4.4,-1.2c0.5,-0.2 1,-0.4 1.4,-0.7c-0.5,-0.9 -0.9,-1.7 -1.2,-2.7c-0.3,-1.3 -0.3,-2.8 0,-4.5s1.2,-3.2 2.4,-4.8c1.2,-1.5 2.9,-3.1 5,-4.4c2.1,-1.3 4.8,-2.5 7.9,-3.4c2.8,-0.8 5.5,-1.3 8,-1.4c2.5,-0.2 4.7,-0.1 6.7,0.3c2,0.4 3.6,1.2 4.9,2.3s2.3,2.5 2.7,4.3c0.3,1.2 0.4,2.5 0.3,3.9c-0.1,1.4 -0.4,3.1 -0.9,4.9c-0.5,1.9 -0.9,3.5 -1,4.9c-0.1,1.3 0,2.5 0.2,3.4c0.3,1.2 0.9,2.1 1.7,2.5c0.9,0.4 1.7,0.5 2.8,0.2c1.4,-0.4 2.8,-1.3 4.3,-2.8c1.4,-1.5 2.6,-3.6 3.5,-6.2c0.9,-2.7 1.4,-5.1 1.4,-7.4C134.3,160.4 134,158.3 133.5,156.3z</item>

    </string-array>

    <integer-array name="os_colors">
        <item>@color/white</item>
        <item>@color/black1</item>
        <item>@color/black1</item>
    </integer-array>
```

**Comment by jaredrummler** [comment]
Most likely, you need to change `app:animatedSvgImageSizeX` and `app:animatedSvgImageSizeY` values. The values should be the same as the width and height in the original SVG `viewBox`. Can you post the original SVG?

**Comment by Vishal12796** [comment]
[os_ai.zip](https://github.com/jaredrummler/AnimatedSvgView/files/355472/os_ai.zip)

**Comment by Vishal12796** [comment]
[os_ai1.zip](https://github.com/jaredrummler/AnimatedSvgView/files/355475/os_ai1.zip)

**Comment by Vishal12796** [comment]
problem is sloved  
the problem is animatedSvgImageSizeX and app:animatedSvgImageSizeY
thank you so much....

**Comment by PGMacDesign** [comment]
I seem to be having a variation of this problem. I followed the steps to utilize animatedSvgImageSizeX and app:animatedSvgImageSizeY with the original dimens and that works, But, but image is just slightly to the left and up in that the edges are cut off by the surrounding box. Is there an off-set value somewhere set by default?
Or is there some marginLeft/ top method that can be used to off-set it? Using the marginLeft android xml option does not seem to move it on my end.
-PGMacDesign

**Comment by samy-baili** [comment]
+1 I got the issue than @PGMacDesign

**Comment by PGMacDesign** [comment]
@samy-baili  sort of, basically, the issue was that I was setting the X and Y values in the XML , but I was not setting them the correct number. 

The xml tags:

```
    lib:animatedSvgImageSizeX="770"
    lib:animatedSvgImageSizeY="508"
```

Need to match the svg as defined in your actual SVG file. 
http://i.imgur.com/wkLaSaV.png

Mine still wasn't centering perfectly so I asked our graphic artist to actually center the SVG using an SVG editor tool, then re-opened it and pulled the width and height you see in that xml code above. 
If you try and pull an image from something that does not have the image centered, it will not center no matter how you try. 

So in short, steps are:
1) Make sure image is centered in svg first. IE, like this:  http://screencloud.net/v/3A8I and not multiple images in the same one (IE like this, http://screencloud.net/v/w36Y )
2) Get the SVG dimensions via by opening up the svg image in a text editor (IE Notepad++) and use the width as the X coord and the height as the Y coord. 
3) And it should run from there as long as the height and width match the xml.

**Comment by samy-baili** [comment]
Thank you very much Patrick, you helped me a lot :) !

**Comment by HenriqueMonte** [comment]
Hi, I just try everything and the AnimatedSvgView does not display on center. What am I doing wrong?
My SVG is: 
`<svg style="width:24px;height:24px" viewBox="0 0 24 24">
    <path fill="#000000" d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2,4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z" />
</svg>`

My AnimatedSvgView:

```
        <com.jrummyapps.android.widget.AnimatedSvgView
            android:id="@+id/ivFinished"
            android:layout_width="104dp"
            android:layout_height="104dp"
            android:layout_below="@+id/tvSaleMessage"
            android:layout_centerHorizontal="true"
            android:layout_gravity="center"
            android:layout_marginTop="32dp"
            app:animatedSvgFillColors="@array/finished_sale_colors"
            app:animatedSvgFillStart="1200"
            app:animatedSvgFillTime="1000"
            app:animatedSvgGlyphStrings="@array/finished_sale_strings"
            app:animatedSvgImageSizeX="24"
            app:animatedSvgImageSizeY="24"
            app:animatedSvgTraceTime="2000"
            app:animatedSvgTraceTimePerGlyph="1000" />

```

The image displays in the top left corner, and only a quarter of it appears. 
![svganim](https://cloud.githubusercontent.com/assets/20558370/21986627/562d0a08-dbe7-11e6-9439-915326b13bd3.png)

My arrays:

```
    <string-array name="finished_sale_strings">
        <item>M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2,4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z</item>
    </string-array>

    <integer-array name="finished_sale_colors">
        <item>@color/colorFlatGreenDark</item>
    </integer-array>

```

Thank you!

**Comment by jaredrummler** [comment]
@HenriqueMonte It looks like this SVG is checkbox-marked-circle-outline from https://materialdesignicons.com I've had problems with path data from those icons too. In the past I have opened the SVG in Adobe Illustrator and copied over the path to a new artboard and the path data was updated and worked correctly. I can't seem to get this SVG working either but I only spent about 5 minutes on it. I'm not sure what is going on and I don't have the time at the moment to investigate. Try using another SVG or recreating it in Adobe Illustrator or similar software.

**Comment by HenriqueMonte** [comment]
@jaredrummler Thank you for your answer!  I spent a couple of hours trying to fix this, and this was the solution. I imported the SVG (yes, it is from https://materialdesignicons.com) to Inkscape. I changed the canvas area from 24 to 96 and export the SVG. The path generated was completely different and now it's working like a charm!
